### PR TITLE
Cooperate with eventlet

### PR DESCRIPTION
**Pull request by stephenc-pace**
# Summary
This function 'flush_queue_to_buffer' has no socket io and thus never yields back to the greenlet event loop. When a large response is combined with a large  number of responses (ie, streaming a large response) it can take a long time to finish the call as it loops through serializing every message, resulting in *everything* blocking until the call finishes.

### CODE REVIEW

**Comment by stephenc-pace** [comment]
@mattbennett 
Another one :D 
Requires build fixes for other PRs so will need to wait until they are merged so I don't have to patch them across again.

**Review by mattbennett — APPROVED** [review]
Ah whoops. With hindsight, can see from the implementation that there's no I/O in this method.

Hope it didn't cause too many problems!

**Pull request merged**
